R/dash.R

```diff
@@ -804,129 +1296,154 @@ Dash <- R6::R6Class(
+                                             local_path = private$asset_map$css,
                                              prefix = self$config$requests_pathname_prefix)
-      } 
+      }
       else {
         css_assets <- NULL
       }
-      
+
       # collect CSS assets from external_stylesheets
-      css_external <- vapply(self$config$external_stylesheets, 
-                             generate_css_dist_html, 
+      css_external <- vapply(self$config$external_stylesheets,
+                             generate_css_dist_html,
                              FUN.VALUE=character(1),
-                             local = FALSE)      
-      
+                             local = FALSE)
+
       # collect JS assets from dependencies
-      # 
-      if (!(is.null(private$scripts))) {
-        scripts_assets <- generate_js_dist_html(href = paste0(private$assets_url_path, names(private$scripts)),
+      #
+      if (!(is.null(private$asset_map$scripts))) {
+        scripts_assets <- generate_js_dist_html(href = paste0(private$assets_url_path, names(private$asset_map$scripts)),
                                                 local = TRUE,
-                                                local_path = private$scripts,
+                                                local_path = private$asset_map$scripts,
                                                 prefix = self$config$requests_pathname_prefix)
       } else {
         scripts_assets <- NULL
       }
 
       # collect JS assets from external_scripts
       scripts_external <- vapply(self$config$external_scripts,
-                                 generate_js_dist_html, 
+                                 generate_js_dist_html,
                                  FUN.VALUE=character(1))
-      
+
       # create tag for favicon, if present
       # other_files_map[names(other_files_map) %in% "/favicon.ico"]
-      if ("/favicon.ico" %in% names(private$other)) {
+      if ("/favicon.ico" %in% names(private$asset_map$other)) {
         favicon <- sprintf("<link href=\"/_favicon.ico\" rel=\"icon\" type=\"image/x-icon\">")
       } else {
         favicon <- ""
       }
 
       # set script tag to invoke a new dash_renderer
       scripts_invoke_renderer <- sprintf("<script id=\"%s\" type=\"%s\">%s</script>",
-                                         "_dash-renderer", 
-                                         "application/javascript", 
+                                         "_dash-renderer",
+                                         "application/javascript",
                                          "var renderer = new DashRenderer();")
-                  
+
       # serving order of CSS and JS tags: package -> external -> assets
       css_tags <- paste(c(css_deps,
                           css_external,
                           css_assets),
-                        collapse = "\n")
-      
+                        collapse = "\n            ")
```

Comment:
Is this whitespace necessary here, and in the following instances of `css_tags`?

Reply:
Yeah, this is an attempt to properly indent the tags.